winml/test/adapter/AdapterSessionTest.cpp

```diff
@@ -0,0 +1,311 @@
+  auto session = CreateUniqueOrtSession(model_path, session_options);
+
+  OrtExecutionProvider* ort_provider;
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionGetExecutionProvider(session.get(), 0, &ort_provider), ort_api);
+  // Test if DML EP method can be called
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->DmlExecutionProviderFlushContext(ort_provider), ort_api);
+}
+
+void RegisterGraphTransformers() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+  winml_adapter_api->SessionRegisterGraphTransformers(session.get());
+}
+
+void RegisterGraphTransformers_DML() {
+  GPUTEST
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+  winml_adapter_api->SessionRegisterGraphTransformers(session.get());
+}
+
+void RegisterCustomRegistry() {
+  IMLOperatorRegistry* registry;
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->CreateCustomRegistry(&registry), ort_api);
+  if (registry) {
+    const auto session_options = CreateUniqueOrtSessionOptions();
+    auto session = CreateUniqueOrtSession(session_options);
+    THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionRegisterCustomRegistry(session.get(), registry), ort_api);
+  }
+}
+
+void RegisterCustomRegistry_DML() {
+  GPUTEST
+  IMLOperatorRegistry* registry;
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->CreateCustomRegistry(&registry), ort_api);
+  WINML_EXPECT_NOT_EQUAL(nullptr, registry);
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionRegisterCustomRegistry(session.get(), registry), ort_api);
+}
+
+void LoadAndPurloinModel(const UniqueOrtSession& session, const std::string& model_path) {
+  com_ptr<WinML::IModel> model;
+  WINML_THROW_IF_FAILED(engine_factory->CreateModel(model_path.c_str(), sizeof(model_path), model.put()));
+
+  Microsoft::WRL::ComPtr<WinML::IOnnxruntimeModel> onnxruntime_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(model->QueryInterface(IID_PPV_ARGS(&onnxruntime_model)));
+  OrtModel* ort_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(onnxruntime_model->DetachOrtModel(&ort_model));
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionLoadAndPurloinModel(session.get(), ort_model), ort_api);
+}
+
+void LoadAndPurloinModel() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+  LoadAndPurloinModel(session, "fns-candy.onnx");
+}
+
+void Initialize() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+
+  com_ptr<WinML::IModel> model;
+  const auto model_path = "fns-candy.onnx";
+  WINML_THROW_IF_FAILED(engine_factory->CreateModel(model_path, sizeof(model_path), model.put()));
+
+  Microsoft::WRL::ComPtr<WinML::IOnnxruntimeModel> onnxruntime_model;
```

Comment:
why switch to WRL? should just pick winrt::com_ptr or WRL::ComPtr for consistency